packages/vite/src/node/plugins/asset.ts

```diff
@@ -76,7 +78,12 @@ export function assetPlugin(config: ResolvedConfig): Plugin {
       }
 
       id = id.replace(urlRE, '$1').replace(/[\?&]$/, '')
-      const url = await fileToUrl(id, config, this)
+      const url = await fileToUrl(id, config, this).catch(() => {
+        config.logger.warnOnce(
+          `\n${id} doesn't exist at build time, it will remain unchanged to be resolved at runtime`
+        )
+        return id
+      })
```

Comment:
if make it open, I think it need to use `build.rollupOptions.external`. if match will call warning, else keep error

Reply:
I don't think this would work though. If the import is transformed as `import foo from '/non-existent.png'`, externalizing it is not correct too as it would result to an incorrect output (and error in runtime). Maybe for case like this we should return the path as is, e.g. `foo ===  '/non-existent.png'`.

Reply:
your mean the below can not match after resolved url `/base/img/aaaSMPTE-color-bars.png`? I think user can add the resolved url by hand.
```js
external: [
    '/img/aaaSMPTE-color-bars.png',
  ],
```

I want to make the non-existent file warn and kept the origin url instead of error if the file matches by `build.rollupOptions.external`.

Reply:
Oh yeah I see your point now. In that case, your PR seems to be doing what I've described. Yeah I think suppressing the warning if the import path matches `build.rollupOptions.external` seems like a fine idea

Reply:
But I see your workaround, it is needed?

Reply:
I'm not sure what workaround you're referring to, but the behaviour I'm thinking is that:

If a file can't be resolved, it'll report a warning, which can be suppressed if the import matches `build.rollupOptions.external`.

Maybe I'm misinterpreting your idea too.